src/fake-timers-src.js

```diff
@@ -376,109 +376,59 @@ function withGlobal(_global) {
         return infiniteLoopError;
     }
 
-    /**
-     * @param {Date} target
-     * @param {Date} source
-     * @returns {Date} the target after modifications
-     */
-    function mirrorDateProperties(target, source) {
-        let prop;
-        for (prop in source) {
-            if (source.hasOwnProperty(prop)) {
-                target[prop] = source[prop];
+    //eslint-disable-next-line jsdoc/require-jsdoc
+    function createDate() {
+        class ClockDate extends NativeDate {
+            /**
+             * @param {number} year
+             * @param {number} month
+             * @param {number} date
+             * @param {number} hour
+             * @param {number} minute
+             * @param {number} second
+             * @param {number} ms
+             * @returns void
+             */
+            // eslint-disable-next-line no-unused-vars
+            constructor(year, month, date, hour, minute, second, ms) {
+                // Defensive and verbose to avoid potential harm in passing
+                // explicit undefined when user does not pass argument
+                if (arguments.length === 0) {
+                    super(ClockDate.clock.now);
+                } else {
```

Comment:
```suggestion
                // Defensive and verbose to avoid potential harm in passing
                // explicit undefined when user does not pass argument
                if (arguments.length === 0) {
                    super(ClockDate.clock.now);
                } else {
```
Maybe cover this with a test? 🤔